src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiManifest.groovy

```diff
@@ -68,35 +78,89 @@ class JpiManifest extends Manifest {
 
-        mainAttributes.putValue('Jenkins-Version', conv.coreVersion)
+        mainAttributes['Jenkins-Version'] = conv.coreVersion
 
-        mainAttributes.putValue('Mask-Classes', conv.maskClasses)
+        mainAttributes['Mask-Classes'] = conv.maskClasses
 
         def dep = findDependencyProjects(project)
         if (dep.length() > 0) {
-            mainAttributes.putValue('Plugin-Dependencies', dep)
+            mainAttributes['Plugin-Dependencies'] = dep
         }
 
         if (conv.pluginFirstClassLoader) {
-            mainAttributes.putValue('PluginFirstClassLoader', 'true')
+            mainAttributes['PluginFirstClassLoader'] = 'true'
         }
 
         if (conv.developers) {
-            mainAttributes.putValue(
-                    'Plugin-Developers',
-                    conv.developers.collect { "${it.name ?: ''}:${it.id ?: ''}:${it.email ?: ''}" }.join(',')
-            )
+            mainAttributes['Plugin-Developers'] = conv.developers.collect {
+                "${it.name ?: ''}:${it.id ?: ''}:${it.email ?: ''}"
+            }.join(',')
         }
 
         YesNoMaybe supportDynamicLoading = isSupportDynamicLoading(classDir)
         if (supportDynamicLoading != YesNoMaybe.MAYBE) {
-            mainAttributes.putValue('Support-Dynamic-Loading', (supportDynamicLoading == YesNoMaybe.YES).toString())
+            mainAttributes['Support-Dynamic-Loading'] = (supportDynamicLoading == YesNoMaybe.YES).toString()
         }
 
         // remove empty values
         mainAttributes.entrySet().removeAll { it.value == null || it.value.toString().empty }
+
+        mainAttributes
+    }
+
+    @Override
+    Map<String, Attributes> getSections() {
+        [:]
+    }
+
+    @Override
+    Manifest attributes(Map<String, ?> map) throws ManifestException {
+        throw new UnsupportedOperationException()
+    }
+
+    @Override
+    Manifest attributes(Map<String, ?> map, String s) throws ManifestException {
+        throw new UnsupportedOperationException()
+    }
+
+    @Override
+    Manifest getEffectiveManifest() {
+        this
+    }
+
+    @Override
+    Manifest writeTo(Writer writer) {
+        throw new UnsupportedOperationException()
+    }
+
+    @Override
+    @SuppressWarnings('Instanceof')
+    Manifest writeTo(Object o) {
+        if (!(o instanceof OutputStream)) {
+            throw new IllegalArgumentException()
+        }
+        toJavaManifest().write(o)
+        this
+    }
+
+    @Override
+    Manifest from(Object... objects) {
+        throw new UnsupportedOperationException()
+    }
+
+    @Override
+    Manifest from(Object o, Closure<?> closure) {
+        throw new UnsupportedOperationException()
+    }
+
+    java.util.jar.Manifest toJavaManifest() {
+        java.util.jar.Manifest manifest = new java.util.jar.Manifest()
+        attributes.each {
+            manifest.mainAttributes.putValue(it.key, it.value.toString())
+        }
+        manifest
```

Comment:
I prefer using an explicit `return` keyword instead of writing a variable on the last line.